ota-plus-web/docs/modules/ROOT/pages/use-case-ecm.adoc

```diff
@@ -0,0 +1,287 @@
+
+You have your vehicle data in a sophisticated system called "CarHQ" which integrates with your customer database called "CustomerHQ". 
+
+Your "CarHQ" system contains detailed information about all the vehicles in your fleet such as model, build date, warranty date and region. It can also tell you information about customers, what vehicle they have purchased and what support contract they're on.
+
+== A new release is announced
+
+Your release manager contacts you and says that they want to update the software on the Engine Control Module (ECM). 
+
+* The goal is to extend the engine idle time as much as possible to allow the Heating, ventilation, and air conditioning (HVAC) to keep the driver comfortable during extreme weather conditions. 
+* The update will override engine shutdown if temperature falls below 7.2 degrees Celsius or above 35 degrees Celsius.
+
+They have the following release conditions:
+
+* The release only applies to editions of the Lannista "Tireon" model built after 2018
+* Restrict the update to the Scandinavian market 
+* Target only customers who have opted for premium support.
+
+== Device groups
+
+=== Preparation
+You can run detailed search queries in CarHQ and export the data to a CSV file. Your search query might look something like this. It's simplified for the purposes of this tutorial.
+
+[source,sql]
+----
+Select VIN
+FROM CustomerView
+WHERE model="Tireon" AND "BuildDate => 2018-01-01" AND Region IN ('DK, NO, SE, FI') AND SupportLevel ="010A"
+----
+
+In this case, you'd want to create a fixed group. 
+
+* A smart group is not sophisticated enough for your query since smart groups can only filter by Device ID.
+* Instead, you can use a fixed group and import the search results that you exported from CarHQ.
+
+
+*To prepare the data for import:*
+
+. You'd export your query to a CSV file. Open the file in Excel and remove all any other columns except the VIN numbers.
+. Save a copy of this file as a TEXT file as "__Tireon-vins.txt__".
+. **Tidy up the file format**: Open the text file in a text editor such as Notepad++ and make sure there are no carriage return characters, and the lines are separated only by new line characters. 
+** In Notepad++ you can click the *Show All Characters* button to show non-printable characters such as tabs and new lines.
+** In Notepad++ you also find carriage returns by searching for the regular expression "`\r`".
+
+// MC: Hack to make this section collapsible
++++ <details><summary> +++
+*What should have happened first* 
++++ </summary><div> +++
+
+To group your vehicles, they must already exist as "devices" in OTA Connect. In other words, your developers should have provisioned these devices with a proper device identifier - in this case we're using the VIN number.
+
++++ </div></details> +++
+
+Make sure that the devices are already provisioned with OTA Connect
+You can check that the devices have been provisioned by doing a couple of spot checks. 
```

Comment:
I know that currently we fail silently for any items in an import list that don't exist. But we really shouldn't, and I really hate documenting around bugs.

Again, since this is a case study, perhaps we could skip this part? And try to get Béné to prioritize fixing the fail-silently bug?

Reply:
My point here is that we should never recommend using a feature in a way that includes a step like "Do the thing, then manually check to see if it worked."

Reply:
yep, fixed - also in 09f8a31c2bdeb578aef8af373fdfb5b3c70af2c4